src/pages/Register.tsx

```diff
@@ -0,0 +1,283 @@
+import { useUniqueId } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import React, { ReactNode, useState } from 'react';
+import { ButtonPrimary, ButtonSecondary, LabeledCheckbox } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import { TextInput } from 'src/components/input';
+import planet from 'src/images/register-planet.svg';
+import { Ajax } from 'src/libs/ajax';
+import { SamUserAttributes } from 'src/libs/ajax/User';
+import { refreshTerraProfile, signOut } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { reportError } from 'src/libs/error';
+import Events from 'src/libs/events';
+import { FormLabel } from 'src/libs/forms';
+import { RegistrationLogo } from 'src/libs/logos';
+import { authStore, getTerraUser, TerraUser } from 'src/libs/state';
+import validate from 'validate.js';
+
+const constraints = (partOfOrg: boolean) => {
+  return {
+    givenName: { presence: { allowEmpty: false } },
+    familyName: { presence: { allowEmpty: false } },
+    email: { presence: { allowEmpty: false } },
+    institute: { presence: { allowEmpty: !partOfOrg } },
+    department: { presence: { allowEmpty: !partOfOrg } },
+    title: { presence: { allowEmpty: !partOfOrg } },
+  };
+};
+
+type CheckboxLineProps = {
```

Comment:
we want to favor interface over type where we can... better type error reporting... only use type when necessary for `Pick<>` or other util type usage